File: medplum-java/medplum-model/src/main/java/com/fhir/ClientApplication.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableClientApplication.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ClientApplication {
  Optional<Meta> meta();

  Optional<String> name();

  String resourceType();

  Optional<code> language();

  Optional<uri> implicitRules();

  Optional<id> id();

  String secret();

  Optional<String> redirectUri();

  Optional<String> description();

  static ImmutableClientApplication.ResourceTypeBuildStage builder() {
    return ImmutableClientApplication.builder();
  }
}

File: medplum-typescript/src/model/fhir/ClientApplication.ts
import { code } from "./code";
import { id } from "./id";
import { Meta } from "./Meta";
import { uri } from "./uri";

export interface ClientApplication {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    name: string | null | undefined;
    description: string | null | undefined;
    secret: string;
    redirectUri: string | null | undefined;
}
